internal/pkg/manifest/worker_svc_test.go

```diff
@@ -410,6 +410,127 @@ func TestWorkerSvc_ApplyEnv(t *testing.T) {
 			},
 		},
 	}
+	duration111Seconds := 111 * time.Second
+	mockWorkerServiceWithSubscribeNilOverride := WorkerService{
+		Workload: Workload{
+			Name: aws.String("phonetool"),
+			Type: aws.String(WorkerServiceType),
+		},
+		WorkerServiceConfig: WorkerServiceConfig{
+			Subscribe: &SubscribeConfig{
+				Topics: &[]TopicSubscription{
+					{
+						Name:    "name",
+						Service: "svc",
+						Queue: &SQSQueue{
+							Name:       aws.String("queue"),
+							Retention:  &duration111Seconds,
+							Delay:      &duration111Seconds,
+							Timeout:    &duration111Seconds,
+							KMS:        aws.Bool(true),
+							DeadLetter: &DeadLetterQueue{Tries: aws.Uint16(10)},
+							FIFO:       &FIFOOrBool{Enabled: aws.Bool(true)},
+						},
+					},
+				},
+			},
+		},
+		Environments: map[string]*WorkerServiceConfig{
+			"test-sub": {
+				Subscribe: nil,
+			},
+		},
+	}
+	mockWorkerServiceWithNilSubscribeOverride := WorkerService{
+		Workload: Workload{
+			Name: aws.String("phonetool"),
+			Type: aws.String(WorkerServiceType),
+		},
+		WorkerServiceConfig: WorkerServiceConfig{
+			Subscribe: nil,
+		},
+		Environments: map[string]*WorkerServiceConfig{
+			"test-sub": {
+				Subscribe: &SubscribeConfig{
+					Topics: &[]TopicSubscription{
+						{
+							Name:    "name",
+							Service: "svc",
+							Queue: &SQSQueue{
+								Name:       aws.String("queue"),
+								Retention:  &duration111Seconds,
+								Delay:      &duration111Seconds,
+								Timeout:    &duration111Seconds,
+								KMS:        aws.Bool(true),
+								DeadLetter: &DeadLetterQueue{Tries: aws.Uint16(10)},
+								FIFO:       &FIFOOrBool{Enabled: aws.Bool(true)},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	mockWorkerServiceWithEmptySubscribeOverride := WorkerService{
+		Workload: Workload{
+			Name: aws.String("phonetool"),
+			Type: aws.String(WorkerServiceType),
+		},
+		WorkerServiceConfig: WorkerServiceConfig{
+			Subscribe: &SubscribeConfig{
+				Topics: &[]TopicSubscription{
+					{
+						Name:    "name",
+						Service: "svc",
+						Queue: &SQSQueue{
+							Name:       aws.String("queue"),
+							Retention:  &duration111Seconds,
+							Delay:      &duration111Seconds,
+							Timeout:    &duration111Seconds,
+							KMS:        aws.Bool(true),
+							DeadLetter: &DeadLetterQueue{Tries: aws.Uint16(10)},
+							FIFO:       &FIFOOrBool{Enabled: aws.Bool(true)},
+						},
+					},
+				},
+			},
+		},
+		Environments: map[string]*WorkerServiceConfig{
+			"test-sub": {
+				Subscribe: &SubscribeConfig{},
+			},
+		},
+	}
+	mockWorkerServiceWithSubscribeEmptyOverride := WorkerService{
+		Workload: Workload{
+			Name: aws.String("phonetool"),
+			Type: aws.String(WorkerServiceType),
+		},
+		WorkerServiceConfig: WorkerServiceConfig{
+			Subscribe: &SubscribeConfig{},
```

Comment:
No need to update your code - just thinking out loud! I wonder what manifest input will give us `&SubscribeConfig{}` for `Subscribe` 🤔  If they don't have `subscribe` in manifest at all i'm assuming it'll just be `nil` (please correct me if this is wrong!).

Reply:
Good point, it looks like it's nil, so I removed that test case since nil is already covered.